app-backend/tile/src/main/scala/directives/TileErrorHandler.scala

```diff
@@ -0,0 +1,41 @@
+package com.azavea.rf.tile
+
+import geotrellis.spark.io.LayerIOError
+import com.azavea.rf.common.RollbarNotifier
+import com.azavea.rf.common.RfStackTrace
+import akka.http.scaladsl.server.{Route, ExceptionHandler, Directives}
+import akka.http.scaladsl.model.StatusCodes
+import com.typesafe.scalalogging.LazyLogging
+import spray.json.{SerializationException, DeserializationException}
+import com.typesafe.scalalogging.LazyLogging
+
+trait TileErrorHandler extends Directives
+    with RollbarNotifier
+    with LazyLogging {
+  val tileExceptionHandler = ExceptionHandler {
+    case e: IllegalArgumentException =>
+      logger.error(RfStackTrace(e))
+      sendError(e)
+      complete(StatusCodes.ClientError(400)("Bad Argument", e.getMessage))
+    case e: IllegalStateException =>
+      logger.error(RfStackTrace(e))
+      sendError(e)
+      complete(StatusCodes.ClientError(400)("Bad Request", e.getMessage))
+    case e: DeserializationException =>
+      logger.error(RfStackTrace(e))
+      sendError(e)
+      complete(StatusCodes.ClientError(400)("Decoding Error", e.getMessage))
+    case e: SerializationException =>
+      logger.error(RfStackTrace(e))
+      sendError(e)
+      complete(StatusCodes.ServerError(500)("Encoding Error", e.getMessage))
+    case e: LayerIOError =>
```

Comment:
Using `LayerIOError` here is not ideal; it is probably overinclusive and may capture some situations where we would want to return a 500. We may want to consider defining our own error hierarchy so we can better express our expected failure conditions.